matchms/networking/SimilarityNetwork.py

```diff
@@ -0,0 +1,135 @@
+
+    """
+    def __init__(self, identifier: str = "spectrumid",
+                 top_n: int = 20,
+                 max_links: int = 10,
+                 score_cutoff: float = 0.7,
+                 link_method: str = 'single'):
+        """
+        Parameters
+        ----------
+        identifier
+            Unique intentifier for each spectrum in scores. Will also be used for
+            node names.
+        top_n
+            Consider edge between spectrumA and spectrumB if score falls into
+            top_n for spectrumA or spectrumB (link_method="single"), or into
+            top_n for spectrumA and spectrumB (link_method="mutual"). From those
+            potential links, only max_links will be kept, so top_n must be >= max_links.
+        max_links
+            Maximum number of links to add per node. Default = 10.
+            Due to incoming links, total number of links per node can be higher.
+        score_cutoff
+            Threshold for given similarities. Edges/Links will only be made for
+            similarities > score_cutoff. Default = 0.7.
+        link_method
+            Chose between 'single' and 'mutual'. 'single will add all links based
+            on individual nodes. 'mutual' will only add links if that link appears
+            in the given top-n list for both nodes.
+        """
+        # pylint: disable=too-many-arguments
+        self.identifier = identifier
+        self.top_n = top_n
+        self.max_links = max_links
+        self.score_cutoff = score_cutoff
+        self.link_method = link_method
+        self.graph = None
+
+    @staticmethod
+    def _select_edge_score(similars_scores: dict, scores_type: numpy.dtype):
+        """Chose one value if score contains multiple values (e.g. "score" and "matches")"""
+        if len(scores_type) > 1 and "score" in scores_type.names:
+            return {key: value["score"] for key, value in similars_scores.items()}
+        if len(scores_type) > 1:  # Assume that first entry is desired score
+            return {key: value[0] for key, value in similars_scores.items()}
+        return similars_scores
+
+    def create_network(self, scores: Scores) -> nx.Graph:
+        """
+        Function to create network from given top-n similarity values. Expects that
+        similarities given in scores are from an all-vs-all comparison including all
+        possible pairs.
+
+        Args:
+        --------
+        scores
+            Matchms Scores object containing all spectrums and pair similarities for
+            generating a network.
+        """
+        assert self.top_n >= self.max_links, "top_n must be >= max_links"
+        assert numpy.all(scores.queries == scores.references), \
+            "Expected symmetric scores object with queries==references"
+        unique_ids = list({s.get(self.identifier) for s in scores.queries})
+
+        # Initialize network graph, add nodes
+        msnet = nx.Graph()
+        msnet.add_nodes_from(unique_ids)
```

Comment:
Yes, that was intentional (since people often keep all elements for their plotting to also show the singletons). 
I now added a `keep_unconnected_nodes` option, which by default is set to True.

Reply:
Thanks